base/util_testing.go

```diff
@@ -760,7 +760,7 @@ func TestRequiresCollections(t *testing.T) {
 	}
 }
 
-func waitUntilScopeAndCollectionExists(collection *gocb.Collection) error {
+func WaitUntilScopeAndCollectionExists(collection *gocb.Collection) error {
```

Comment:
Nitpick: This is no longer a testing utility - can it be moved to `base/collection.go`?

Might be nice to have a comment describing why we have to do the Exists check in the first place too.

```suggestion
// WaitUntilScopeAndCollectionExists will try to perform an `Exists` operation on an arbitrary doc ID in the given scope and collection until it can succeed in telling us whether the doc exists or not.
// There's no WaitForReady operation in GoCB for collections, so attempting to use the collection in this way this seems like our best option.
func WaitUntilScopeAndCollectionExists(collection *gocb.Collection) error {
```